### PR TITLE
Replace vpa-exporter with kube-state-metrics

### PR DESCRIPTION
**Pull request by istvanballok**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:

Replace vpa-exporter with kube-state-metrics

Previously, the vpa-exporter component was used to expose VPA related metrics.
With this PR, the vpa-exporter component is no longer used and equivalent (but differently named) metrics are exposed by the existing kube-state-metrics deployments. The Grafana dashboards are adjusted and aligned appropriately.

This PR also contains a related network bandwidth usage improvement: previously the control plane prometheus-es used to scrape the kube-state-metrics independently, getting metrics for the whole seed cluster. Now they only federate the relevant metrics from the cache prometheus.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

Please check the individual commits.

To facilitate code review and to verify the impact of the source code changes at runtime, the second half of the commits of this PR, starting with [this](https://github.com/gardener/gardener/pull/6771/commits/f7076152149b208081056f3cabc93941aeb415f2?diff=unified&w=0) commit, explains the effects that this PR introduces in the KinD based local setup of Gardener with a single shoot cluster.

Here is a screenshot of the revised dashboard:
image

/cc @rickardsjp @wyb1 

**Release note**:

```other operator
Replace vpa-exporter with kube-state-metrics.
The vpa-exporter is no longer used in Gardener.
The kube-state-metrics component is exposing the VPA related metrics.
```

### CODE REVIEW

**Review by wyb1 — APPROVED** [review]

**Review by rickardsjp — APPROVED** [review]

File: pkg/operation/botanist/component/vpa/monitoring.go [hunk 1]
```diff
@@ -15,91 +15,17 @@
...
-	}); err != nil {
-		return component.CentralMonitoringConfig{}, err
-	}
-
-	return component.CentralMonitoringConfig{ScrapeConfigs: []string{scrapeConfig.String()}}, nil
+	return component.CentralMonitoringConfig{ScrapeConfigs: []string{}}, nil
 }
 
 // ScrapeConfigs returns the scrape configurations for Prometheus.
 func (v *vpa) ScrapeConfigs() ([]string, error) {
```
Comment (rfranzke) [inline pkg/operation/botanist/component/vpa/monitoring.go]: It seems this file is useless now. Can you delete it?

File: pkg/utils/images/images.go [hunk 2]
```diff
@@ -113,8 +113,6 @@ const (
 	ImageNameTelegraf = "telegraf"
 	// ImageNameVpaAdmissionController is a constant for an image in the image vector with name 'vpa-admission-controller'.
 	ImageNameVpaAdmissionController = "vpa-admission-controller"
-	// ImageNameVpaExporter is a constant for an image in the image vector with name 'vpa-exporter'.
-	ImageNameVpaExporter = "vpa-exporter"
```
Comment (rfranzke) [inline pkg/utils/images/images.go:117]: Please also delete it from `charts/images.yaml`.

File: charts/seed-monitoring/charts/grafana/dashboards/operators/vpa-dashboard.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
   "editable": true,
   "gnetId": null,
   "graphTooltip": 0,
-  "iteration": 1655446204243,
```
Comment (wyb1) [inline charts/seed-monitoring/charts/grafana/dashboards/operators/vpa-dashboard.json:18]: The dashboard should be moved to the `owners` folder

**Review by rfranzke — COMMENTED** [review]
Nice PR! 👏🏻

File: charts/seed-monitoring/charts/grafana/dashboards/operators/vpa-dashboard.json (hunk 3 above)
Comment (istvanballok) [reply charts/seed-monitoring/charts/grafana/dashboards/operators/vpa-dashboard.json:18]: Done.

File: pkg/utils/images/images.go (hunk 2 above)
Comment (istvanballok) [reply pkg/utils/images/images.go:117]: Thank you for catching this! The change in the `charts/images.yaml` was somehow lost in a rebase. :(

File: pkg/operation/botanist/component/vpa/monitoring.go (hunk 1 above)
Comment (istvanballok) [reply pkg/operation/botanist/component/vpa/monitoring.go]: Thank you for this review comment! We deleted the file and the VPA component no longer implements the `MonitoringComponent` interface.

**Comment by istvanballok** [comment]
Hi @wyb1 and @rfranzke,
@rickardsjp and I applied your review suggestions, updated the description and added a few explanatory commits, please take another look.

cc @voelzmo The revised VPA dashboard could provide some insights how the VPA works in our seed clusters.

**Comment by istvanballok** [comment]
Pending tasks:

- [x] hide max/upper series by default
- [x] execute make generate

**Review by rickardsjp — APPROVED** [review]

**Review by rfranzke — COMMENTED** [review]
Very nice, well done!

/lgtm
/approve

**Pull request merged**